### ISSUE TITLE
Edge.predicate should be required?

### ISSUE THREAD

**Issue by edeutsch** [NCATSTranslator/ReasonerAPI#376]
Richard pointed out that:

```
    Edge:
      type: object
      description: >-
        A specification of the semantic relationship linking two concepts
        that are expressed as nodes in the knowledge "thought" graph
        resulting from a query upon the underlying knowledge source.
      properties:
        predicate:
          allOf:
            - $ref: '#/components/schemas/BiolinkPredicate'
          nullable: true
...
        attributes:
          type: array
          description: A list of additional attributes for this edge
          items:
            $ref: '#/components/schemas/Attribute'
          nullable: true
...
      required:
        - subject
        - object
```

i.e. `predicates` is NOT currently required. Shouldn't it be?

Should attributes be required, too? (if we think every Edge should at least have provenance information)

**Comment by sierra-moxon** [comment]
+1 for attributes being required.

**Comment by edeutsch** [comment]
General feeling at the call was that attributes should be nullable, but that the semantic validator should emit a warning when and Edge has no attributes.

**Comment by edeutsch** [comment]
Addressed in PR #377
